### PR TITLE
Release

### PR DESCRIPTION
**Pull request by stevefalco**
I have completed my testing.  You should be able to pull my release branch into kicad/master.

The next nightly build will have the libraries split off into separate rpm files.  If you want to preview what this looks like, you can visit my copr repo:

https://copr.fedorainfracloud.org/coprs/stevenfalco/kicad/build/722286/

### CODE REVIEW

**Comment by nickoe** [comment]
Thank you. This needs to be rebased because of @aimylios' earlier changes was merged before you began your work. I prefer if there are no merge commits such that it can be fast forwarded or rebased cleanly. If you need any help with this please speak up.

File: builder.sh [hunk 1]
```diff
@@ -155,6 +155,7 @@ make_tar "kicad-packages3D"	"${FULL_VERSION}"	"${TAG}"
 if [ "${TAG}" != "HEAD" ]; then
 	# Building for release.
 	sed \
+		-e "s/@EPOCH@/1/" \
```
Comment (nickoe) [inline builder.sh]: What is the purpose of this commit?

**Comment by aimylios** [comment]
I think this work is a good base for the discussion I want to start in #7, but I would advise against merging this as is, even after a rebase. It is difficult to follow all the modifications, but as far as I can see this PR includes changes to the SPEC file I made in my [release branch](https://github.com/aimylios/fedora-kicad-packaging/tree/release) that are completely unrelated to the split-up into several packages (like enabling new build options and cleanup of dependencies). Let us proceed in small steps, so that the next generation of contributors can understand what we did and why we did it.

Also, I would prefer if the nightlies would not change the packaging strategy too often, so if you merge this (or any other proposal) then that should ideally be compatible to what upstream does or at least very close to that.

**Comment by stevefalco** [comment]
I'm surprised that it needed a rebase.  At the time I sent my pull request github was telling me it was ready for merge as a fast forward.

I retract the patch, since it seems like there is an interest in moving slower, and that there are other issues.

Perhaps the best course of action is to wait until @aimylios completes all the spec changes, at which point we can consider revisiting rewriting the scripts.

**Comment by stevefalco** [comment]
The idea of

` +		-e "s/@EPOCH@/1/" \`

is that if one is using my scripts to make an -rc build, then you want the epoch to be 1, but if you are using my scripts to make a nightly build, then you want the epoch to be larger, so the nightly takes priority over the -rc.  At least that is the way I understand it from @aimylios .

**Comment by stevefalco** [comment]
Regarding being compatible with upstream, I've made contact with the Fedora package maintainer, and I could send him a spec file to show what is planned.  But since that is now in need of further discussion, I don't know what to send him.  It seems like we may be in a "catch 22" situation.

Perhaps the best way to proceed is to get @lkundrak involved in the discussions here, rather than trying to go back and forth between the two teams.  I'll suggest that in an email to him.

**Comment by aimylios** [comment]
@nickoe
The Fedora KiCad package uses an Epoch of 1. That means all other stable builds need the Epoch to be set to 1 as well, or the upstream package will try to "update" (i.e. overwrite) them every time you do a `dnf update`. As stevefalco already explained, the nightlies need a higher Epoch (currently a value of 100 is used) because they use a version number which is incompatible to stable builds.

@stevefalco
In general, I think your changes are OK, we should just not do everything at once. The latest Copr build failed because of the integration of the doc subpackage. Apparently Fedora 26 needs a special treatment. And that was a fairly simple modification. I plan to submit a separate PR for a little more cleanup and to enable OCE and Python scripting. After that, only the split-up is left (as far as I can see).

**Pull request closed without merging**